platform/ios/scripts/generate-style-code.js

```diff
@@ -0,0 +1,82 @@
+'use strict';
+
+const fs = require('fs');
+const ejs = require('ejs');
+const spec = require('mapbox-gl-style-spec').latest;
+
+global.camelize = function (str) {
+    return str.replace(/(?:^|-)(.)/g, function (_, x) {
+        return x.toUpperCase();
+    });
+}
+
+global.camelizeWithLeadingLowercase = function (str) {
+    return str.replace(/-(.)/g, function (_, x) {
+        return x.toUpperCase();
+    });
+}
+
+global.isObject = function (property) {
+    switch (property.type) {
+        case 'string':
+            return true;
+        case 'color':
+            return true;
+        case 'array':
+            return true;
+        case 'enum': // (enum will be converted to NSString hence isObject(enum) == true)
+            return true;
+        default:
+            return false;
+    }
+}
+
+global.propertyType = function (property) {
+    switch (property.type) {
+        case 'boolean':
+            return 'BOOL';
+        case 'number':
+            return 'float';
```

Comment:
CGFloat seems to suit every number I've seen so far (width, padding, spacing, angle, blur, size) Let's start with that and see if we find any edge cases we should handle differently.